cli/src/generate/grammar_files.rs

```diff
@@ -97,16 +98,21 @@ pub fn generate_grammar_files(
             let mut package_json = serde_json::from_str::<Map<String, Value>>(&package_json_str)
                 .with_context(|| "Failed to parse package.json")?;
             let package_json_main = package_json.get("main");
+            let package_json_types = package_json.get("types");
             let package_json_needs_update = package_json_main.map_or(true, |v| {
```

Comment:
I think this old check should be removed. Instead we should add checks for the other newly-added or changed fields. That's out of scope for this PR but if you want to handle it you can.

Reply:
I might handle it in a followup PR that I'm creating that adds prebuild, assuming this one get's merged.

Reply:
Personally, I'd prefer `prebuildify` over `prebuild` (ideally without `node-gyp-build`). Most parser binaries are not particularly large so bundling them would be fine. Worst case, we can limit the prebuilds to the major platforms (win-x64, linux-x64, mac-x64, mac-arm64). The [workflow](https://github.com/tree-sitter/workflows/blob/main/.github/workflows/package-npm.yml) can be adapted accordingly.

Reply:
> I think this old `main` check should be removed. Instead we should add checks for `files` and `dependencies` (`nan` -> `node-addon-api`). That's out of scope for this PR but if you want to handle it you can.

Just did this